apm-agent-plugins/apm-servlet-plugin/src/main/java/co/elastic/apm/agent/servlet/ServletTransactionHelper.java

```diff
@@ -146,30 +146,35 @@ public static void setUsernameIfUnset(@Nullable String userName, TransactionCont
 
     public String normalizeServletPath(String requestURI, @Nullable String contextPath, @Nullable String servletPath, @Nullable String pathInfo) {
         String path = servletPath;
-        if (path == null || path.isEmpty()) {
-            logger.debug("Weblogic fallback applied. requestURI = {}, contextPath = {}, servletPath = {}, pathInfo = {}", requestURI, contextPath, servletPath, pathInfo);
-
-            int start = 0;
-            int end = requestURI.length();
-            if (contextPath != null && contextPath.length() > 0) {
-                if (!contextPath.equals("/")) {
-                    start = contextPath.length();
-                }
-            }
-            if (pathInfo != null) {
-                end -= pathInfo.length();
-            }
+        if (path != null && !path.isEmpty()) {
+            return path;
+        }
 
-            if (start == end) {
-                // use complete request URI instead of an empty string as fallback
-                path = requestURI;
-            } else {
-                path = requestURI.substring(start, end);
+        logger.debug("Empty servlet path fallback applied. requestURI = {}, contextPath = {}, servletPath = {}, pathInfo = {}", requestURI, contextPath, servletPath, pathInfo);
```

Comment:
nit: I'd guard this with a if so that the varargs array is only allocated when the debug log level is enabled.